p2p/conn/secret_connection.go

```diff
@@ -98,9 +111,14 @@ func MakeSecretConnection(conn io.ReadWriteCloser, locPrivKey crypto.PrivKey) (*
 	if err != nil {
 		return nil, err
 	}
+	io.ReadFull(hkdfHiEphPub, hdkf_state[:])
+
+	hkdfSecret := hkdf.New(hash, dhSecret[:], hdkf_state[:], []byte("TENDERMINT_SECRET_CONNECTION_TRANSCRIPT_HASH"))
+
+	io.ReadFull(hkdfSecret, hdkf_state[:])
 
 	// generate the secret used for receiving, sending, challenge via hkdf-sha2 on dhSecret
-	recvSecret, sendSecret, challenge := deriveSecretAndChallenge(dhSecret, locIsLeast)
+	recvSecret, sendSecret, challenge := deriveSecretAndChallenge(hdkf_state, locIsLeast)
```

Comment:
As far as I understand, the changes make some kind of transcript hashing (similar to what noise does). As far as I can tell this is an improvement to the protocol. 

It would be cool, if there was a test that shows how this previously failed and is now fixed. Ideally, this would come with some formal reasoning, too. This is even more so important as we do not have any test-vectors to compare against (?).

Reply:
`bump` could we add a test, it can also live with the p2p tests `make test_p2p`